keiko-spring/src/main/kotlin/com/netflix/spinnaker/config/QueueConfiguration.kt

```diff
@@ -95,11 +98,14 @@ class QueueConfiguration {
   fun enabledActivator(queueProperties: QueueProperties) = EnabledActivator(queueProperties.enabled)
 
   @Bean
-  fun queueEventPublisher(
-    applicationEventPublisher: ApplicationEventPublisher
-  ) = object : EventPublisher {
-    override fun publishEvent(event: QueueEvent) {
-      applicationEventPublisher.publishEvent(event)
-    }
-  }
+  @ConditionalOnMissingBean(EventPublisher::class)
+  fun queueEventPublisher() = NoopEventPublisher()
+
+  @Bean
+  @ConditionalOnProperty("queue.metrics.enabled", havingValue = "true", matchIfMissing = true)
+  fun queueMetricsPublisher(
+    registry: Registry,
+    clock: Clock,
+    queue: MonitorableQueue
+  ) = QueueMetricsPublisher(registry, clock, queue)
```

Comment:
that's really the only change to the environment, we swap out the `ApplicationEventPublisher` for a `QueueMetricsPublisher` and bypass Spring events for direct metrics. I thought this was pretty neat.

Reply:
Yep, this is how I was hoping you'd do it.